components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManagerTest.java

```diff
@@ -403,4 +414,62 @@ public AddReceiptResponse addConsent(ReceiptInput receiptInput) throws ConsentMa
             return null;
         }
     }
+
+    @Test(dataProvider = "userDetailsForTestEmailOTPVerification")
+    public void testEmailOTPVerification(String username, String userstore, String tenantDomain,
+                                         String preferredChannel, String errorMsg,
+                                         String enableInternalNotificationManagement, String enableEmailOTP, String expectedChannel)
+            throws Exception {
+
+        // Build recovery user.
+        User user = new User();
+        user.setUserName(username);
+        user.setUserStoreDomain(userstore);
+        user.setTenantDomain(tenantDomain);
+
+        mockConfigurations("true", enableEmailOTP, enableInternalNotificationManagement);
+
+        String secretKey = Utils.generateSecretKey(preferredChannel,tenantDomain, String.valueOf(RecoveryScenarios.SELF_SIGN_UP));
+
+        UserSelfRegistrationHandler handler = new UserSelfRegistrationHandler();
+
+        UserRecoveryData userRecoveryData = new UserRecoveryData(user, secretKey, RecoveryScenarios
+                .SELF_SIGN_UP, RecoverySteps.CONFIRM_SIGN_UP);
+        // Storing preferred notification channel in remaining set ids.
+        userRecoveryData.setRemainingSetIds(preferredChannel);
+
+        mockJDBCRecoveryDataStore(userRecoveryData);
+        mockEmailTrigger();
+    }
+
+    @DataProvider(name = "userDetailsForTestEmailOTPVerification")
+    private Object[][] userDetailsForTestEmailOTPVerification() {
+
+        String username = "test-user";
+        // Notification channel types.
+        String EMAIL = NotificationChannels.EMAIL_CHANNEL.getChannelType();
+        String SMS = NotificationChannels.SMS_CHANNEL.getChannelType();
+        String EXTERNAL = NotificationChannels.EXTERNAL_CHANNEL.getChannelType();
+
+        /* ArrayOrder: Username, Userstore, Tenant domain, Preferred channel, Error message, Manage notifications
+        internally, excepted channel */
+        return new Object[][]{
+                {username, TEST_USERSTORE_DOMAIN, TEST_TENANT_DOMAIN_NAME, EMAIL, "User with EMAIL as Preferred " +
```

Comment:
if we have the same `username`, `TEST_USERSTORE_DOMAIN`, `TEST_TENANT_DOMAIN_NAME`, they are not required to be passed from the data provider. They can be defined in the test method itself

Reply:
There was a similar implementation for a test in lines 212 - 221. I just followed the same implementation for this one as well.